faillint/faillint_test.go

```diff
@@ -223,3 +245,75 @@ func TestRun(t *testing.T) {
 		})
 	}
 }
+
+func TestHasDirective(t *testing.T) {
+	type input struct {
+		comments []*ast.Comment
+		option   string
+	}
+	type expected struct {
+		out     bool
+		message string
+	}
+	for _, tcase := range []struct {
+		name     string
+		input    input
+		expected expected
+	}{
+		{
+			name: "missing reason on ignore",
+			input: input{
+				comments: []*ast.Comment{
+					{Text: "//faillint:ignore"},
+				},
+				option: ignoreKey,
+			},
+			expected: expected{
+				out:     true,
+				message: fmt.Sprintf(missingReasonTemplate, "ignore"),
+			},
+		},
+		{
+			name: "missing reason on file-ignore",
+			input: input{
+				comments: []*ast.Comment{
+					{Text: "//faillint:file-ignore"},
+				},
+				option: fileIgnoreKey,
+			},
+			expected: expected{
+				out:     true,
+				message: fmt.Sprintf(missingReasonTemplate, "file-ignore"),
+			},
+		},
+		{
+			name: "file-ignore when ignore expected",
+			input: input{
+				comments: []*ast.Comment{
+					{Text: "//faillint:file-ignore"},
+				},
+				option: ignoreKey,
+			},
+			expected: expected{
+				out:     false,
+				message: fmt.Sprintf(unexpectedFileIgnoreTemplate, "file-ignore"),
+			},
+		},
```

Comment:
I think it would be nice to have also cases with `reason` and are valid so we can see it works for those as well. Right now seems like we only check for failures but not for successes.

Reply:
We were covering success in the "integration tests" but it's trivial to add valid cases here and I will do so.

Reply:
Done in https://github.com/fatih/faillint/pull/11/commits/33d8aa80b077c51ca044b1f39ba1416f106bfdfd.